src/merklize.rs

```diff
@@ -120,32 +125,69 @@ impl Iterator for MerklizedChunks {
 impl MerklizedChunks {
 	/// Compute `MerklizedChunks` from a list of erasure chunks.
 	pub fn compute(chunks: Vec<Vec<u8>>) -> Self {
-		let mut hashes: Vec<Hash> = chunks
-			.iter()
-			.map(|chunk| {
+		let chunks_len = chunks.len();
+		let target_size = chunks_len.next_power_of_two();
+		
+		// Parallel chunk hashing
+		#[cfg(feature = "parallel")]
```

Comment:
May use conditional macro/trait/function,
if the only difference is iterator type (`.iter()` vs `.par_iter()`).
```rust
macro_rules! feature_par_iter {
    ($x:expr) => ({
        #[cfg(feature="parallel")]
        let iter = $x.par_iter();
        #[cfg(not(feature="parallel"))]
        let iter = $x.iter();
        iter
    })
}
```